packages/svelte/messages/compile-errors/template.md

```diff
@@ -192,6 +192,14 @@
 
 > %thing% is invalid inside <%parent%>
 
+HTML has some restrictions where certain elements can appear. For example, a `<div>` inside a `<p>` is invalid. Some violations "only" result in invalid HTML, others result in the HTML being repaired by the browser, resulting in content shifting around. Some examples:
+
+- `<p>hello <div>world</div></p>` will result in `<p>hello </p><div>world</div><p></p>` for example (the `<div>` autoclosed the `<p>`)
+- `<option><div>option a</div></option>` will result in `<option>option a</option>` (the `<div>` is removed)
+- `<table><tr><td>cell</td></tr></table>` will result in `<table><tbody><tr><td>cell</td></tr></tbody></table>` (a `<tbody>` is auto-inserted)
+
+Svelte throws a compiler error when it detects that it will generate the HTML in such a way that it will always be repaired and result in the runtime code not finding the nodes at the expected locations.
```

Comment:
I think we can shrink this a bit:

```suggestion
HTML restricts where certain elements can appear. For example, a `<p>` element cannot contain another `<p>`, or a `<div>`, or any other block-level element. The browser will 'repair' the HTML in a way that breaks Svelte's assumptions about the structure of your components. Some more examples:

- `<option><div>...</div></option>` will result in `<option>...</option>` (everything except text is removed)
- `<table><tr><td>...</td></tr></table>` will result in `<table><tbody><tr><td>...</td></tr></tbody></table>` (a `<tbody>` is auto-inserted)
```

I'm curious about this:

> Some violations "only" result in invalid HTML

What are those violations? Are they relevant in this context? Are we doing anything to prevent them? If not then I don't think we need to mention them

Reply:
```suggestion
HTML restricts where certain elements can appear. In case of a violation the browser will 'repair' the HTML in a way that breaks Svelte's assumptions about the structure of your components. Some examples:

- `<p>hello <div>world</div></p>` will result in `<p>hello </p><div>world</div><p></p>` for example (the `<div>` autoclosed the `<p>` because `<p>` cannot contain block-level elements)
- `<option><div>option a</div></option>` will result in `<option>option a</option>` (the `<div>` is removed)
- `<table><tr><td>cell</td></tr></table>` will result in `<table><tbody><tr><td>cell</td></tr></tbody></table>` (a `<tbody>` is auto-inserted)

Svelte throws a compiler error when it detects that it will generate the HTML in such a way that it will always be repaired and result in the runtime code not finding the nodes at the expected locations.
```

The last sentence is because there's a warning version of this, and so they provide context on why one throws while the other is a warning.

Reply:
I think the last sentence is unnecessary tbh — we've just explained why the HTML is invalid, so why _wouldn't_ we throw an error? I reckon we can get away with just explaining why we're _not_ throwing an error on the warning message

Reply:
Ok let's do away with the last sentence - updated my previous suggestion